api/src/main/java/com/messagebird/objects/conversations/MessageComponentType.java

```diff
@@ -10,8 +13,18 @@ public enum  MessageComponentType {
   FOOTER("footer"),
   BUTTON("button"),
   CARD("card"),
-  CAROUSEL("carousel");
+  CAROUSEL("carousel"),
+  LIMITED_TIME_OFFER("limited_time_offer");
 
+  private static final Map<String, MessageComponentType> TYPE_MAP;
+
+  static {
+    Map<String, MessageComponentType> map = new HashMap<>();
+    for (MessageComponentType componentType : MessageComponentType.values()) {
+      map.put(componentType.getType().toLowerCase(), componentType);
+    }
+    TYPE_MAP = Collections.unmodifiableMap(map);
```

Comment:
if we use java 9+ we can have TYPE_MAP = Map.copyOf(map);

Reply:
Thanks for the great suggestion to use Map.copyOf(map)! It’s a cleaner and more efficient way to create an unmodifiable map, especially with Java 9 and higher. However, since this library is used by multiple external companies, many of whom may still be running Java 8 in their environments, upgrading the source and target version to Java 10 would introduce compatibility issues for them.

To maintain broad compatibility with these users, I think it’s best for now to stick with Java 8 to ensure the library continues to work seamlessly in a wide range of environments.